contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -755,9 +763,11 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     }
 
     /**
-     * @dev Returns the next message ID to be used to send a message to the given `originBlockchainID`.
+     * @dev Calculates the message ID given this contract's address and required input parameters.
      */
-    function _getNextMessageID(bytes32 originBlockchainID) private view returns (uint256) {
-        return latestMessageIDs[originBlockchainID] + 1;
+    function _calculateNextMessageID() private view returns (bytes32) {
+        bytes32 blockchainID_ = blockchainID;
+        require(blockchainID_ != bytes32(0), "TeleporterMessenger: blockchainID not set");
+        return sha256(abi.encode(address(this), blockchainID_, messageNonce));
```

Comment:
What's the point of a `private view` function, is it some sort of gas optimization?

Reply:
It took me a minute to remember why we have it structured this way 😅 

The reason is that interface functions (such as `getNextMessageID`) are required to be `external`. In order to call those methods from within the contract itself, you have to use `this.functionName()`, which uses significantly more gas, and is generally not best practice. 

So instead, we have a `private view` method for internal use and add on `getNextMessageID` which provides the external access point to it.